syntax = "proto3";

message FileData {
  string filename = 1;
  bytes data = 2;
}

message UploadResponse {
  string message = 1;
}

message DownloadRequest {
  string fileLocation = 1;
}

message DownloadResponse {
  bytes data = 1;
}

message DeleteRequest {
  string fileLocation = 1;
}

message DeleteResponse {
  string message = 1;
}




service FileTransfer {
  rpc UploadFile(FileData) returns (UploadResponse);
  rpc DownloadFile(DownloadRequest) returns (DownloadResponse);
  rpc DeleteFile (DeleteRequest) returns (DeleteResponse);
}